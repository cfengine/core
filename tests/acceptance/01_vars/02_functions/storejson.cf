# Test that the storejson() function will convert a data container back to JSON

body common control
{
      inputs => { "../../default.cf.sub" };
      bundlesequence => { default("$(this.promise_filename)") };
      version => "1.0";
}

bundle agent test
{
  vars:
      "mylist" slist => { "x", "y" };
      "data" data => parsejson('{
  "x": [
    1,
    2,
    3
  ]
}');

      "datas" string => storejson(data);

      "data_from_slist" string => storejson(mylist);
      "data_from_inline" string => storejson('{ "a": "b" }');
}

#######################################################

bundle agent check
{
  methods:
      "check"  usebundle => dcs_check_state(test,
                                           "$(this.promise_filename).expected.json",
                                           $(this.promise_filename));
}
