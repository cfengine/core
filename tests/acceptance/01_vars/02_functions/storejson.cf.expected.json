{
  "data": {
    "x": [
      1,
      2,
      3
    ]
  },
  "data_from_inline": "{\n  \"a\": \"b\"\n}",
  "data_from_slist": "[\n  \"x\",\n  \"y\"\n]",
  "datas": "{\n  \"x\": [\n    1,\n    2,\n    3\n  ]\n}",
  "mylist": [
    "x",
    "y"
  ]
}
